block.go

```diff
@@ -0,0 +1,52 @@
+package sdk
+
+import (
+	"context"
+
+	ctypes "github.com/cometbft/cometbft/rpc/core/types"
+	cosmos "github.com/cosmos/cosmos-sdk/client"
+)
+
+// NodeStatusFetcher returns the status of the node to which it is connected.
+//
+// This status is used here in the BlockClient in order to get the latest block height.
+type NodeStatusFetcher interface {
+	Status(ctx context.Context) (*ctypes.ResultStatus, error)
+}
+
+// BlockClient is the interface to interact with the on-chain block module.
+//
+// For example, it can be used to get the latest block height.
+//
+// It uses the StatusClient interface of the cometbft to pull the latest block height.
+// This is done to avoid fetching the entire latest block just to extract the block height.
+//
+// The implementations of this interface could leverage websockets to get notified
+// about new blocks and cache the latest block height to avoid querying the blockchain
+// for it every time.
+type BlockClient struct {
```

Comment:
WHy is this struct exposed?

`s/BlockClient/blockClient` and users can import the interface

Reply:
Please review my comment below which explains the motivation behind this.